music21/analysis/harmonicFunction.py

```diff
@@ -0,0 +1,305 @@
+                    ):
+    '''
+    Takes a Roman numeral and returns a corresponding harmonic function label.
+
+    >>> rn1 = roman.RomanNumeral('VI', 'a')
+    >>> fn1 = analysis.harmonicFunction.romanToFunction(rn1)
+    >>> fn1
+    <HarmonicFunction.SUBDOMINANT_MINOR_PARALLELKLANG_MAJOR>
+
+    This can be converted into a string:
+
+    >>> str(fn1)
+    'sP'
+
+    Optionally, set onlyHauptHarmonicFunction to True to return
+    a simplified version with only the HauptHarmonicFunction
+    (one of t, T, s, S, d, D: major and minor forms of the tonic, subdominant and dominant).
+
+    >>> fn1 = analysis.harmonicFunction.romanToFunction(rn1, onlyHauptHarmonicFunction=True)
+    >>> fn1
+    <HarmonicFunction.SUBDOMINANT_MINOR>
+
+    >>> str(fn1)
+    's'
+
+    Inversions are not currently considered (they may be in a future version of this).
+    This function simply uses the romanNumeral attribute of the roman.RomanNumeral object.
+    This excludes inversions, but
+    includes, where applicable, the frontAlterationAccidental.modifier.
+
+    >>> rn2 = roman.RomanNumeral('bII6', 'g')
+    >>> fn2 = analysis.harmonicFunction.romanToFunction(rn2)
+    >>> fn2
+    <HarmonicFunction.SUBDOMINANT_MINOR_GEGENKLANG_MAJOR>
+
+    >>> str(fn2)
+    'sG'
+
+    See further notes on the complementary
+    :func:`~music21.analysis.harmonicFunction.functionToRoman`.
+    '''
+
+    referenceTuples = functionFigureTuplesMajor
+    if rn.key:  # RomanNumeral object can be created without one.
+        if rn.key.mode == 'minor':
+            referenceTuples = functionFigureTuplesMinor
+
+    for thisKey, thisValue in referenceTuples.items():
+        if rn.romanNumeral == thisValue:
+            if onlyHauptHarmonicFunction:
+                return HarmonicFunction(str(thisKey)[0])
+            else:
+                return thisKey
+
+    return False
+
+
+# ------------------------------------------------------------------------------
+
+class Test(unittest.TestCase):
+
+    def testAllFunctionLabelsInEnum(self):
+        '''
+        Test that all the entries in the functionFigureTuples
+        (both major and minor) are represented in the HarmonicFunction enum.
+
+        Also tests one fake (invalid) function label.
+        '''
+
+        # All and only valid
+        for thisHarmonicFunction in functionFigureTuplesMajor:
+            HarmonicFunction(thisHarmonicFunction)
+        for thisHarmonicFunction in functionFigureTuplesMinor:
+            HarmonicFunction(thisHarmonicFunction)
+
+        # Invalid
+        fakeExample = 'TPG'
+        self.assertRaises(ValueError, HarmonicFunction, fakeExample)
+
+    def testFunctionToRoman(self):
+        self.assertEqual(functionToRoman('T').figure, 'I')
+
+    def testSimplified(self):
+        rn = roman.RomanNumeral('III', 'f')
+        self.assertEqual(str(romanToFunction(rn)), 'tP')
+        self.assertEqual(str(romanToFunction(rn, onlyHauptHarmonicFunction=True)), 't')
```

Comment:
Casting to `str` should be unnecessary, since a `StrEnum` already compares equal with a string. Maybe one of the tests could compare with the enum value e.g. `assertIs(..., HarmonicFunction.GEGENKLANG)`.

Reply:
Good shout re including a comparison with the enum value. Thanks!